packages/beacon-node/src/network/core/networkCore.ts

```
@@ -253,25 +253,28 @@ export class NetworkCore implements INetworkCore {
   /** Destroy this instance. Can only be called once. */
   async close(): Promise<void> {
     if (this.closed) return;
-
-    this.clock.off(ClockEvent.epoch, this.onEpoch);
-
-    // Must goodbye and disconnect before stopping libp2p
-    await this.peerManager.goodbyeAndDisconnectAllPeers();
-    this.logger.debug("network sent goodbye to all peers");
-    await this.peerManager.close();
-    this.logger.debug("network peerManager closed");
-    await this.gossip.stop();
-    this.logger.debug("network gossip closed");
-    await this.reqResp.stop();
-    await this.reqResp.unregisterAllProtocols();
-    this.logger.debug("network reqResp closed");
-    this.attnetsService.close();
-    this.syncnetsService.close();
-    await this.libp2p.stop();
-    this.logger.debug("network lib2p closed");
-
-    this.closed = true;
+    try {
+      this.clock.off(ClockEvent.epoch, this.onEpoch);
+
+      // Must goodbye and disconnect before stopping libp2p
+      await this.peerManager.goodbyeAndDisconnectAllPeers();
+      this.logger.debug("network sent goodbye to all peers");
+      await this.peerManager.close();
+      this.logger.debug("network peerManager closed");
+      await this.gossip.stop();
+      this.logger.debug("network gossip closed");
+      await this.reqResp.stop();
+      await this.reqResp.unregisterAllProtocols();
+      this.logger.debug("network reqResp closed");
+      this.attnetsService.close();
+      this.syncnetsService.close();
+      await this.libp2p.stop();
+      this.logger.debug("network lib2p closed");
+
+      this.closed = true;
+    } catch (err) {
+      this.logger.error("Error on NetworkCore.close()", {}, err as Error);
+    }
```

Why are we not throwing the error here, how is it ensured that everything is closed in that case?

There is an explicit exit in the beacon handler but this is only executed if there is an error
https://github.com/ChainSafe/lodestar/blob/e42d6cc6457051cf67ab7a3cdbfada25028d10f7/packages/cli/src/cmds/beacon/handler.ts#L141-L142

---

This code execute when normal exit does not work. And `node` actually could not closed gracefully, that cause this issue. So not sure how else we can cleanup.

---

My point is just if you swallow the error here it won't execute the explicit handler, hence process could hang.

Note: the other process.exit(0) we have should ideally be removed if libp2p active handles issue is resolved

---

I don't think this should be merged without a rationale / explanation for wrapping this in a try-catch and not propagating the errors up to caller.

- Did anything change when in libp2p that makes this necessary now?
- Why does our code not handle that correctly upstream?

IMO this try-catch might hang the process if there is an error in the network shutdown sequence (see above comments) and we should not hide it in this PR (bump libp2p) without having a clear reason for adding it.